src/autoscaler/cf/uaa_client.go

```diff
@@ -0,0 +1,43 @@
+package cf
+
+import (
+	"os"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/cloudfoundry-incubator/uaago"
+)
+
+type UaaClient interface {
+	RefreshAuthToken() (string, error)
+}
+
+type uaaClient struct {
+	logger      lager.Logger
+	conf        *CFConfig
+	uaaClient   *uaago.Client
+	UAAEndpoint string
+}
+
+func NewUaaClient(conf *CFConfig, logger lager.Logger, uaaEndpoint string) UaaClient {
+	uc := &uaaClient{}
+	uc.logger = logger
+	uc.conf = conf
+	uc.UAAEndpoint = uaaEndpoint
+
+	uaaClient, err := uaago.NewClient(uaaEndpoint)
+	if err != nil {
+		uc.logger.Error("failed-to-create-uaago-client", err)
+		os.Exit(1)
+	}
+	uc.uaaClient = uaaClient
+
+	return uc
+}
+
+func (uc *uaaClient) RefreshAuthToken() (string, error) {
+	token, err := uc.uaaClient.GetAuthToken(uc.conf.ClientID, uc.conf.Secret, uc.conf.SkipSSLValidation)
```

Comment:
autoscaler has an implementation of request auth token in 
https://github.com/cloudfoundry-incubator/app-autoscaler/blob/master/src/autoscaler/cf/client.go#L124-L156
which will do post request to "/oauth/token" to get granted as well.  

Why we need to do a uaa_client.go here?

Reply:
Followed  noaa samples suggestion  https://github.com/cloudfoundry/noaa/blob/master/samples/token_refresher/main.go 

I will have a look if we can remove this through a separate PR

Reply:
It looks like there is no reference to this piece of code now.

Reply:
This is being used while creating the firehose consumer to read the metrics.